plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/LinstorStoragePool.java

```diff
@@ -0,0 +1,195 @@
+package com.cloud.hypervisor.kvm.storage;
+
+import java.util.List;
+import java.util.Map;
+
+import com.cloud.storage.Storage;
+import org.apache.cloudstack.utils.qemu.QemuImg;
+
+public class LinstorStoragePool implements KVMStoragePool {
+    private final String _uuid;
+    private final String _sourceHost;
+    private final int _sourcePort;
+    private final Storage.StoragePoolType _storagePoolType;
+    private final StorageAdaptor _storageAdaptor;
+    private final String _resourceGroup;
+
+    public LinstorStoragePool(
+            String uuid,
+            String host,
+            int port,
+            String resourceGroup,
+            Storage.StoragePoolType storagePoolType,
+            StorageAdaptor storageAdaptor) {
+        _uuid = uuid;
+        _sourceHost = host;
+        _sourcePort = port;
+        _storagePoolType = storagePoolType;
+        _storageAdaptor = storageAdaptor;
+        _resourceGroup = resourceGroup;
+    }
+
+    @Override
+    public KVMPhysicalDisk createPhysicalDisk(
+        String name,
+        QemuImg.PhysicalDiskFormat format,
+        Storage.ProvisioningType provisioningType,
+        long size)
+    {
+        return this._storageAdaptor.createPhysicalDisk(
+            name,
+            this,
+            format,
+            provisioningType,
+            size
+        );
+    }
+
+    @Override
+    public KVMPhysicalDisk createPhysicalDisk(String volumeUuid, Storage.ProvisioningType provisioningType, long size)
+    {
+        return this._storageAdaptor.createPhysicalDisk(
+            volumeUuid,
+            this,
+            getDefaultFormat(),
+            provisioningType,
+            size
+        );
+    }
+
+    @Override
+    public boolean connectPhysicalDisk(String volumeUuid, Map<String, String> details)
+    {
+        return this._storageAdaptor.connectPhysicalDisk(volumeUuid, this, details);
+    }
+
+    @Override
+    public KVMPhysicalDisk getPhysicalDisk(String volumeUuid)
+    {
+        return this._storageAdaptor.getPhysicalDisk(volumeUuid, this);
```

Comment:
I think, no need to explicitly mention _this_  for class method / variable unless one exists with same name in that scope.